src/library_tty.js

```diff
@@ -80,6 +91,9 @@ mergeInto(LibraryManager.library, {
             throw new FS.ErrnoError(ERRNO_CODES.EIO);
           }
         }
+        // Sending null causes output to be printed, even if it didn't end in a
+        // newline.
+        stream.tty.ops.put_char(stream.tty, null);
```

Comment:
I'm not sure I follow why this was added? Was there something that did not work before?

Reply:
Oops, I'd meant to be remove that!  It's left over from the previous commit, when I was sending each character through 'put_char' and signalling I'd finished by sending a null.  Since I've changed it to call printChars further up, this is no longer needed.  I'll take it out.